core/src/authorizations.ts

```diff
@@ -180,3 +181,15 @@ export const getDebtAuctionAuthorizationStatus = async (network: string, walletA
     const authorizationStatus = await contract.can(walletAddress, flopperAddress);
     return authorizationStatus.toNumber() === 1;
 };
+
+export const giveAllowanceToAddress = async (
+    network: string,
+    collateralSymbol: string,
+    spenderAddress: string,
+    collateralAmount: BigNumber
+) => {
+    const config = getCollateralConfigBySymbol(collateralSymbol);
+    const tokenAddress = await getContractAddressByName(network, collateralSymbol);
+    const tokenContract = await getErc20Contract(network, tokenAddress, true);
+    await tokenContract.approve(spenderAddress, collateralAmount.shiftedBy(config.decimals).toFixed());
```

Comment:
Before sending amounts to a chain, you always need to have integer-like string without floating points. In case `collateralAmount` was calculated (multiplied/divided) it might have more precision than required, so it's safer to:
```suggestion
    await tokenContract.approve(spenderAddress, collateralAmount.shiftedBy(config.decimals).toFixed(0));
```

Reply:
having precision incorrectly handled on the previous steps results in the error (incorrect argument) here if i remember correctly. And from my perspective it's desired to have the error raised to fix it asap. At least from my experience: having a couple of toFixed(0) where not actually needed already resulted into some time spent debugging

Reply:
> And from my perspective it's desired to have the error raised to fix it asap

I'm just saying that calling this function with specific _appropriate_ parameters will fail. You usually don't want your code to be more fragile to help you with debugging. Also, if this code will be called outside tests, no one would knew your intention to make it fragile. Therefore, please handle this edge case properly and prefer `toFixed(0)` over `.toFixed()` when sending requests to blockchain in the future.

Reply:
- Well, my point is derived from the axiom where the parameters that you would like to support with `(0)` are inappropriate:
  - it implicitly changes the provided value if it is incorrect
  - it is an approval contract: we're trying to approve an amount that does not match the input.
- As far as i am concerned this actually "declines" incorrect inputs. Code should fail on incorrect inputs / be fragile enough to crash & burn on incorrect inputs.

If the function is called outside the tests with inappropriate arguments in `()` case - it will fail because the input is not correct. In the `(0)` noone will know that it changes the allowance amount without explicit intent of the function user

Introduced the change though.

Reply:
> As far as i am concerned this actually "declines" incorrect inputs

See my initial comment

> In case collateralAmount was calculated (multiplied/divided) it might have more precision than required

Let's say for some reason, you need to get the approval for the 1 ETH divided by 3. There is better edge-case that actually already happen to me, the precision of DAI stored in your VAT is [RAD (ie 45 decimal points)](https://github.com/makerdao/dss/blob/fa4f6630afb0624d04a003e920b0d71a00331d98/src/vat.sol#L59) while the precision of DAI token itself is only 18. So it you need to authorize some number taken from VAT, you will end up with the precision grater than 18 and the transaction will fail. Is this enough to convince you?

Reply:
This comment was not addressed

Reply:
I've created the issue myself https://github.com/sidestream-tech/unified-auctions-ui/issues/517